pkg/asset/manifests/azure/cluster.go

```diff
@@ -296,5 +296,5 @@ func getNextAvailableIP(ctx context.Context, installConfig *installconfig.Instal
 			}
 		}
 	}
-	return "", fmt.Errorf("failed to get available IP in given machine network: %w", err)
+	return "", fmt.Errorf("failed to get available IP in given machine network: this error may be caused by lack of necessary permissions")
```

Comment:
question: I think here `*ipAvail.AvailableIPAddresses` is non-empty and we could not find any suitable IP right? How about `no available IP` or similar?

Reply:
There is IP available but at this stage there's no Ip that we can use that's in the machine network so I think it's fine but then I am really bad at error messages or commenting.

Reply:
Right, what do you think of using what you just wrote as error message instead (i.e. that is `There is IP available but at this stage there's no Ip that we can use that's in the machine network`)? Otherwise, LGTM.

Reply:
I realized the wordings in my original question was kinda odd. Your latest commit looks great! 👏🏼